### ISSUE TITLE
bad code generated if class has underscores in it (e.g., PAYMENT_TYPE)

### ISSUE THREAD

**Issue by The-Alchemist** [immutables/immutables#154]
...the enum and the ``@Value.Immutable`` are in different packages (forcing the code to have an 'import')

e.g.,

## The Enum

```java
package b;

public enum PAYMENT_TYPE {
    PAYPAL,
    CREDIT_CARD,
    BITCOIN;

}

```

## The @Value.Immutable class

```java
package a;

import org.immutables.value.Value;

import b.PAYMENT_TYPE;

@Value.Immutable
public interface SomeValue {
    PAYMENT_TYPE getPaymentType();
}
```

## Generated Code: ImmutableSomeValue

```java
package a;

import b.PAYMENT;
import com.google.common.base.MoreObjects;
import com.google.common.base.Preconditions;
import com.google.common.collect.Lists;
import java.util.Collection;
import javax.annotation.Generated;
//...........snip.................
```

NOTE that the ``import`` is bad:  
``import b.PAYMENT`` instead of ``import b.PAYMENT_TYPE`` 

I think this also happens if it's a regular class and not just an enum.

**Comment by elucash** [comment]
Thank you for the report! It's definitely something with the imports postprocessor.

**Comment by The-Alchemist** [comment]
@elucash : thanks!  if you point me in the right direction, i can submit a pull request.

**Comment by elucash** [comment]
My first guess is that it's improper reading of source import, starting from
[PostprocessingMachine.java#L147](https://github.com/immutables/immutables/blob/master/generator/src/org/immutables/generator/PostprocessingMachine.java#L147)

Overall idea of import processing is that if we can interpret/rewrite things, we do this, if we do not support something - we leave it as is, apparently we are dropping part of an identifer somewhere while we should copy it as is.

**Comment by augustotravillio** [comment]
Not quite there, but near
@elucash can you take a look at #155 ?

**Comment by The-Alchemist** [comment]
Submitted a pull request (https://github.com/immutables/immutables/pull/156)

**Comment by elucash** [comment]
released in 2.0.18, thanks!
